### PR TITLE
Person

### PR DESCRIPTION
**Pull request by maria-salmasian**
This is a project called person it has 3 subclasses programmer, dancer and singer.
the project comes with an UML diagram.

### CODE REVIEW

File: ProjectPerson/src/Dancer.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+public class Dancer extends Person {
```
Comment (hhov88) [inline ProjectPerson/src/Dancer.java]: @maria-salmasian please add comments

File: ProjectPerson/src/Programmer.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private String getCompanyName() {
+        return companyName;
+    }
+
+    private void coding() {
+        System.out.println(getName() + " is a senior developer , at " + getCompanyName());
+    }
+
+    @Override
+    protected void walk(int steps) {
```
Comment (hhov88) [inline ProjectPerson/src/Programmer.java]: @maria-salmasian I think this functions can move to parent class without overriding in all child classes

Comment (maria-salmasian) [reply ProjectPerson/src/Programmer.java]: right! fixed it

File: ProjectPerson/src/Dancer.java (hunk 1 above)
Comment (maria-salmasian) [reply ProjectPerson/src/Dancer.java]: added comments

File: ProjectPerson/src/Dancer.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    private String danceType;
+
+
+    /**
+     * constructor for dancer
+     * @param name
+     * @param designation
+     * @param groupName
+     */
+    protected Dancer(String name, String designation, String groupName) {
```
Comment (hhov88) [inline ProjectPerson/src/Dancer.java]: @maria-salmasian why constructor is protected ?

File: ProjectPerson/src/Person.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    protected void setStep(int steps) {
+        this.steps = steps;
+    }
+
+
+    /**
+     * returning the name of person
+     * @return
+     */
+    protected String getName() {
```
Comment (hhov88) [inline ProjectPerson/src/Person.java]: @maria-salmasian getter and setter can be public

**Review by nsimonyan — APPROVED** [review]

**Pull request merged**